### PR TITLE
feat: add connection popover

### PR DESCRIPTION
**Pull request by PabloRN**
General description: This pull request is mostly an UI/UX improvement to the connection functionality.

Description: When we want to connect the dashboard to an endpoint it needs some configuration. If some of the configuration setup are wrong the system will shows a popup to reconnect or to go to the documentation. This is not very user friendly as we will need to read the whole documentation to understand that we need to go to the settings to setup the whole config, then go back to the log stream and try again. 

Proposal: We can setup basic connection configs from the same connection component (host and port)

### CODE REVIEW

**Review by imsergiy — APPROVED** [review]

File: src/components/Layout/MainNavbar/MainNavbar.tsx [hunk 1]
```diff
@@ -29,8 +29,8 @@ type Props = {
   toggleSidebar?: () => void;
   reconnect?: () => void;
```
Comment (BastinJafari) [inline src/components/Layout/MainNavbar/MainNavbar.tsx:30]: reconnect is not getting used anymore, so it should also be removed from props, no?

**Review by BastinJafari — CHANGES_REQUESTED** [review]

File: src/components/Layout/MainNavbar/MainNavbar.tsx (hunk 1 above)
Comment (PabloRN) [reply src/components/Layout/MainNavbar/MainNavbar.tsx:30]: Agree. I removed it from the child and not from the parent component.

File: src/components/Layout/MainNavbar/ConnectionIndicator.tsx [hunk 2]
```diff
@@ -1,33 +1,136 @@
-import React from "react";
+import React, { useState, useEffect, useRef } from "react";
 // @ts-ignore
 import { NavItem, NavLink, Badge } from "shards-react";
+import { Popover, Overlay } from "react-bootstrap";
+import { Row, Col, Button, Form } from "shards-react";
+import { Store, Dispatcher, Constants } from "../../../flux";
+import BaseFormTextInput from "components/Common/BaseFormTextInput";
```
Comment (BastinJafari) [inline src/components/Layout/MainNavbar/ConnectionIndicator.tsx:7]: Could you change the name of the default export in the file BaseFormTextInput to the file name?

**Review by BastinJafari — CHANGES_REQUESTED** [review]
It would be nice, if you open an issue before implementing a feature and then linking it to that issue. There are infos on that in the contributing guidelines:

https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md

It would also be good to write a describtion into your pull request.
What does this pull request do, why does it make sense, etc.

**Comment by PabloRN** [comment]
> It would be nice, if you open an issue before implementing a feature and then linking it to that issue. There are infos on that in the contributing guidelines:
> 
> https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md
> 
> It would also be good to write a describtion into your pull request.
> What does this pull request do, why does it make sense, etc.

Hello, @BastinJafari 
Thx for the replay.
About to create an issue related to the feature. I didn't see any info in the contributing doc and for my understanding not always a feature is resolving an issue. In this case is more an improvement as the system can work without it. It will be more an UI/UX improvement that will make easy for users to use the system. What do think?
About the description I totally agree. I will do it. Thx again.

File: src/components/Layout/MainNavbar/ConnectionIndicator.tsx (hunk 2 above)
Comment (PabloRN) [reply src/components/Layout/MainNavbar/ConnectionIndicator.tsx:7]: Yes sure. Done. Thank you

**Comment by BastinJafari** [comment]
Hey Pablo, could you rebase your PR?
We had to get rid of some broken checks.

**Comment by PabloRN** [comment]
> Hey Pablo, could you rebase your PR?
> We had to get rid of some broken checks.

Hey @BastinJafari,
The sync with master is done but still a falling case. Caused by redux it seems.
`11:02:16 AM: /opt/build/repo/src/redux/index.ts
11:02:16 AM: TypeScript error in /opt/build/repo/src/redux/index.ts(1,10):
11:02:16 AM: Module '"."' has no exported member 'combineReducers'. Did you mean to use 'import combineReducers from "."' instead?  TS2614`

**Comment by BastinJafari** [comment]
Hey @PabloRN, we are currently transitioning from flux to redux. Could you update your implementation to that? You can see a reference in my newest PR

**Comment by PabloRN** [comment]
> Hey @PabloRN, we are currently transitioning from flux to redux. Could you update your implementation to that? You can see a reference in my newest PR

Sure! Let me give a quick look and I will back to you

**Pull request closed without merging**